File: backend/src/models/DLOAIMATHANG.ts
import { Model, DataTypes } from 'sequelize';
import db from '../config/database';

class DLOAIMATHANG extends Model {
    public id!: any;
    public name!: any;
    public note!: any;
    public status!: any;
    public usermodifiedid!: any;
    public timemodified!: any;
    public timecreated!: any;
    public sortorder!: any;
    public usercreatedid!: any;
    public parentid!: any;
    public parentdir!: any;
    public itemtype!: any;
    public autoid!: any;
    public simageid!: any;
    public cobanhang!: any;
    public cotonkho!: any;
    public codinhluong!: any;
}

DLOAIMATHANG.init({
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true, autoIncrement: true,
    },
    NAME: {
        type: DataTypes.STRING,
        
    },
    NOTE: {
        type: DataTypes.STRING,
        
    },
    STATUS: {
        type: DataTypes.BOOLEAN,
        
    },
    USERMODIFIEDID: {
        type: DataTypes.INTEGER,
        
    },
    TIMEMODIFIED: {
        type: DataTypes.DATE,
        
    },
    TIMECREATED: {
        type: DataTypes.DATE,
        
    },
    SORTORDER: {
        type: DataTypes.INTEGER,
        
    },
    USERCREATEDID: {
        type: DataTypes.INTEGER,
        
    },
    PARENTID: {
        type: DataTypes.INTEGER,
        
    },
    PARENTDIR: {
        type: DataTypes.STRING,
        
    },
    ITEMTYPE: {
        type: DataTypes.STRING,
        
    },
    AUTOID: {
        type: DataTypes.INTEGER,
        
    },
    SIMAGEID: {
        type: DataTypes.INTEGER,
        
    },
    COBANHANG: {
        type: DataTypes.STRING,
        
    },
    COTONKHO: {
        type: DataTypes.STRING,
        
    },
    CODINHLUONG: {
        type: DataTypes.STRING,
        
    },
}, {
    sequelize: db.sequelize,
    modelName: 'DLOAIMATHANG',
    tableName: 'DLOAIMATHANG',
    timestamps: false
});

export default DLOAIMATHANG;

File: frontend/Models/DLOAIMATHANG.cs
using System;

namespace QuanLyBar.Client.Models
{
    public class DLOAIMATHANG
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Note { get; set; }
        public bool? Status { get; set; }
        public string UsermodifiedId { get; set; }
        public DateTime? Timemodified { get; set; }
        public DateTime? Timecreated { get; set; }
        public string Sortorder { get; set; }
        public string UsercreatedId { get; set; }
        public string ParentId { get; set; }
        public string Parentdir { get; set; }
        public string Itemtype { get; set; }
        public int? AutoId { get; set; }
        public string SimageId { get; set; }
        public string Cobanhang { get; set; }
        public string Cotonkho { get; set; }
        public string Codinhluong { get; set; }

        // UI helper properties
        public System.Windows.Media.ImageSource IconSource { get; set; }
        public bool IsBanHang => Cobanhang == "30" || Cobanhang == "1" || Cobanhang?.ToLower() == "true";
        public bool IsTonKho => Cotonkho == "30" || Cotonkho == "1" || Cotonkho?.ToLower() == "true";
        public bool IsDinhLuong => Codinhluong == "30" || Codinhluong == "1" || Codinhluong?.ToLower() == "true";
    }
}
